packages/core/src/zip-foundry.ts

```diff
@@ -0,0 +1,223 @@
+          [
+            `instance = new ${c.name}();`,
+          ],
+          '}',
+        ],
+        getContractSpecificTestFunction(),
+      ),
+      '}',
+    ];
+  }
+
+  function getContractSpecificTestFunction(): Lines[] {
+    if (opts !== undefined) {
+      switch (opts.kind) {
+        case 'ERC20':
+        case 'ERC721':
+          return [
+            'function testName() public {',
+            [
+              `assertEq(instance.name(), "${opts.name}");`
+            ],
+            '}',
+          ];
+
+        case 'ERC1155':
+          return [
+            'function testUri() public {',
+            [
+              `assertEq(instance.uri(0), "${opts.uri}");`
+            ],
+            '}',
+          ];
+
+        case 'Governor':
+        case 'Custom':
+          return [
+            'function testSomething() public {',
+            [
+              '// Add your test here',
+            ],
+            '}',
+          ]
+
+        default:
+          throw new Error('Unknown ERC');
+      }
+    }
+    return [];
+  }
+};
+
+const script = (c: Contract) => {
+  return formatLinesWithSpaces(
+    2,
+    ...spaceBetween(
+      getHeader(c),
+      getImports(c),
+      getScript(c),
+    ),
+  );
+
+  function getImports(c: Contract) {
+    return [
+      'import "forge-std/Script.sol";',
+      `import "../src/${c.name}.sol";`,
+    ];
+  }
+
+  function getScript(c: Contract) {
+    return [
+      `contract ${c.name}Script is Script {`,
+      spaceBetween(
+        [
+          'function setUp() public {}',
+        ],
+        [
+          'function run() public {',
+          [
+            'vm.startBroadcast();',
+            `${c.name} instance = new ${c.name}();`,
+            'console.log("Contract deployed to %s", address(instance));',
+            'vm.stopBroadcast();',
+          ],
+          '}',
+        ],
+      ),
+      '}',
+    ];
+  }
+};
+
+const setupSh = `\
+# Check if git is installed
+if ! [ -x "$(command -v git)" ]
+then
+  echo "git command not found. Install git and try again."
+  exit 1
+fi
+
+# Check if Foundry is installed
+if ! [ -x "$(command -v forge)" ]
+then
+  echo "forge command not found. Install Foundry and try again. See https://book.getfoundry.sh/getting-started/installation"
+  exit 1
+fi
+
+# Setup Foundry project
+if ! [ -f "foundry.toml" ]
+then
+  echo "Initializing Foundry project..."
+
+  # Initialize sample Foundry project
+  forge init --force --no-commit --quiet
+
+  # Install OpenZeppelin Contracts
+  forge install OpenZeppelin/openzeppelin-contracts@v${contracts.version} --no-commit --quiet
+
+  # Remove template contracts
+  rm src/Counter.sol
+  rm script/Counter.s.sol
+  rm test/Counter.t.sol
+
+  # Add remappings
+  if [ -f "remappings.txt" ]
+  then
+    echo "" >> remappings.txt
+  fi
+  echo "@openzeppelin/=lib/openzeppelin-contracts/" >> remappings.txt
+
+  # Perform initial git commit
+  git add .
+  git commit -m "Initial commit" --quiet
```

Comment:
`forge init` already generates initial commits. This kind of message seems more in line with those commits:

```suggestion
  git commit -m "openzeppelin: add wizard output" --quiet
```

I don't like it fully but I didn't know what else to put before the colon?

Reply:
Currently we are using `forge init` with `--no-commit`.

The challenge is Wizard output is already in the directory before we run `forge init`.  So the initial commit would include the Wizard output anyways, regardless of when we do the commit.

We also need to remove template contracts, and add remappings.txt.  I consider it neatest to commit everything at the same time.

(The alternative would be something like this, but it feels excessive:
commit 1: forge init along with Wizard output (let forge init do the commit)
commit 2: remove templates
commit 3: install OZ contracts and add remappings)

Reply:
I agree with changing the commit message format to be more similar to what `forge` normally does though.

Reply:
Ah, I wasn't thinking of all that. It makes sense to use `--no-commit` and generate a single commit. I would consider perhaps adding some mention of wizard in the initial commit message, that's all.